client/src/main/java/io/pravega/client/connection/impl/TcpClientConnection.java

```diff
@@ -159,7 +159,9 @@ static WireCommand readCommand(InputStream in, IoBuffer buffer) throws IOExcepti
         }
         
         public void stop() {
-            stop.set(true);
+            if (stop.getAndSet(true)) {
```

Comment:
Should the get and set separated out here? 
* This method returns if `get` `stop` returns true. 
* Set is called after the callback is complete, using compareAndSet(false, true). 

This might be especially useful if `callback.connectionDropped() can fail with an exception. 

On the other hand, if the failure of `connectionDropped()` call can be safely discounted, the existing code'd be OK.

Reply:
yeah, the idea is the to ensure that we do not stop and already stopped/closed connection.